File: Mixter.Domain/Identity/Session.cs
using System;
using Mixter.Domain.Identity.Events;

namespace Mixter.Domain.Identity
{
    [Aggregate]
    public class Session
    {
        [Command]
        public static void LogIn(IEventPublisher eventPublisher, UserId userId)
        {
            var id = SessionId.Generate();
            eventPublisher.Publish(new UserConnected(id, userId, DateTime.Now));
        }
    }
}

File: src/main/scala/domain/identity/Session.scala
package domain.identity

import domain.EventPublisher
import domain.identity.event.{UserConnected, UserDisconnected}

object Session {
  private case class DecisionProjection(userConnected: UserConnected){
    val userId: UserId = userConnected.userId
    val sessionId: SessionId = userConnected.sessionId
  }
}

case class Session(userConnected:UserConnected){
  private val projection=Session.DecisionProjection(userConnected)

  def logout()(implicit ep:EventPublisher):Unit =
    ep.publish(UserDisconnected(projection.sessionId, projection.userId))
}
